Simplenote/src/main/java/com/automattic/simplenote/utils/SearchTokenizer.java

```diff
@@ -94,10 +97,26 @@ private String parseQuery() {
                 continue;
             }
 
+            if (current == HYPHEN
+                    && last != SPACE
+                    && position != 0
+                    && (!inStrictTerm || quoteChar == SINGLE_QUOTE)) {
```

Comment:
```suggestion
            if (current == HYPHEN && last != SPACE && position != 0) {
```

Since there is a `continue` statement above this which occurs when `inStrictTerm` is `true`, `!inStrictTerm` will always be `true` at this point.  Therefore, `!inStrictTerm || quoteChar == SINGLE_QUOTE` will always be `true`.  So, we can remove that condition.